omnipaxos/src/omni_paxos.rs

```diff
@@ -73,18 +69,77 @@ impl OmniPaxosConfig {
     }
 }
 
-impl Default for OmniPaxosConfig {
+/// Configuration for an `OmniPaxos` cluster.
+/// # Fields
+/// * `configuration_id`: The identifier for the cluster configuration that this Sequence Paxos replica is part of.
+/// * `nodes`: The nodes in the cluster i.e. the `pid`s of the other replicas in the configuration.
+/// * `initial_leader`: The initial leader of the cluster. Could be used in combination with reconfiguration to skip the prepare phase when switching to a new configuration.
+#[derive(Clone, Debug, PartialEq)]
+#[cfg_attr(any(feature = "serde", feature = "toml_config"), derive(Deserialize))]
+#[cfg_attr(feature = "toml_config", serde(default))]
+#[cfg_attr(feature = "serde", derive(Serialize))]
+pub struct ClusterConfig {
+    /// The identifier for the cluster configuration that this Sequence Paxos replica is part of.
+    pub configuration_id: u32,
+    /// The nodes in the cluster i.e. the `pid`s of the other replicas in the configuration.
+    pub nodes: Vec<u64>,
```

Comment:
Use `NodeId` instead of u64

```suggestion
    pub nodes: Vec<NodeId>,
```